text/0195-ranking-score.md

```diff
@@ -0,0 +1,183 @@
+| `attribute`            | <ul><li>`attribute_ranking_order_score`: Results sorted based on the attribute ranking order</li><li>`query_word_distance_score`: Documents with attributes containing the query words close to their position in the query will be considered more relevant than documents containing the query words far from their position in the query</li></ul>                                                    |
+| `exactness`            | <ul><li>`matchType`: It has one of the following values:<ul><li>`exactMatch`: The query exactly matches the entire value of an attribute</li><li>`matchesStart`: The query matches exactly the start of the value of an attribute</li><li>`noExactMatch`: The query doesn't exactly match a document </li></ul></li><li>`matchingWords`: for `matchesStart`, the number of exact words contained in an attribute. The higher the better</li><li>`maxMatchingWords`: for `noExactMatch`, the maximum number of exact words contained in an attribute</li></ul> |
+
+#### 3.2.3. Sort ranking rules
+
+`Sort` and `geosort` ranking rules appear as fields in the score details, but with the following difference:
+
+- Their key follows the following format: `{:attribute-sorted-on}:{:sort-direction}`, with the `:attribute-sorted-on` the name of the attribute that is being sorted on, and the `:sort-direction` either `asc` if the sort is in ascending order, or `desc` if the sort is in descending order. For the `geosort` ranking rule, it is similarly `_geoPoint({:lat}, {:lng}):{:sort-direction}`, with the `:lat` and `:lng` being the latitude and respective longitude of the point that serves as base to sort by distance.
+- They don't have a `score` field, but instead they have a `value` field, representing the value used to sort the document. It is typically the value of the sorted attribute for the document, but can sometimes be a subvalue (case where the value is an array of values).
+- For the `geosort`, there is an additional `distance` field representing the distance between the target point and the point used in the document to sort the document.
+
+#### 3.2.4 Example
+
+The following is an example of a `_scoreDetails` returned for a document matching a search query.
+
+
+```json
+"_rankingScoreDetails": {
+    "words": {
+        "order": 0,
+        "matchingWords": 1,
+        "maxMatchingWords": 1,
+        "score": 1
+    },
+    "typo": {
+        "order": 1,
+        "typoCount": 0,
+        "maxTypoCount": 1,
+        "score": 1
+    },
+    "proximity": {
+        "order": 2,
+        "score": 1
+    },
+    "attribute": {
+        "order": 3,
+        "attributes_ranking_order": 0.8333333333333334,
```

Comment:
Should it be written `attributesRankingOrder` to follow the camelCase field naming convention ?

Reply:
Yea Louis was onto me today about this. It looks like the implementation is outputting it with underscores rather than camelCase so will need a change there first (and then an update here and in the blog post to make sure they are the same). I'll create an issue on the project as a start point.

Reply:
Issues created here: https://github.com/meilisearch/meilisearch/issues/3941

Reply:
Hey many 👋👶

attributeRankingOrderScore and queryWordDistanceScore

[Relevant discussion](https://meilisearch.slack.com/archives/C05A9SRPMFB/p1687350573869669?thread_ts=1687268496.920209&cid=C05A9SRPMFB) (internal link)